Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
+    command.commands
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      options do
+        cmd_name "commands"
+        desc "Show a list of built-in and external commands."
+        option "quiet", desc: "list only the names of commands without the header." do
+          option "include-aliases", desc: "the aliases of internal commands will be included."
```

Comment:
@raza15 Yes, that dimension of detail in `--help` is lost in the second example. My suggestion is to deal with that as a need for documentation, rather than code. In other words, I think we should aim for `--help` output like this:

```
brew commands [options]

Valid options:
  --foo  foo description
  --bar  bar description (requires --foo)
  --baz  baz descroption (requires --bar)
```

Where each of those descriptions comes from the `desc:` string, *not* from the code itself.